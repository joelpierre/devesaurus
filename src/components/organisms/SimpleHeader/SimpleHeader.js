import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';

import Hamburger from '../../atoms/Hamburger/Hamburger';
import styles from './SimpleHeader.module.scss';
import * as coreActions from '../../../store/actions';
import MenuItem from '../../atoms/MenuItem/MenuItem';

export class SimpleHeader extends PureComponent {
  constructor() {
    super();
    this.toggleMenu = this.toggleMenu.bind(this);
  }

  /**
   * Toggle the menu state
   * @param e
   */
  toggleMenu(e) {
    e.preventDefault();

    const { isMenuOpen, setMenuState } = this.props;

    setMenuState(!isMenuOpen);
  }

  render() {
    const { isMenuOpen } = this.props;

    return (
      <header
        className={classNames([styles['simple-header']])}
        data-test="component-simple-header"
      >
        <div className={classNames(styles['simple-header__wrapper'])}>
          <Hamburger
            className={styles['simple-header__hamburger']}
            descriptor="Menu"
            isMenuOpen={isMenuOpen}
            onClick={this.toggleMenu}
          />

          <nav className={styles['simple-header__menu']}>
            <ul className={classNames(styles['simple-header__list'])}>
              <MenuItem
                item={{
                  url: '/about',
                  className: styles['simple-header__link'],
                }}
                className={classNames(styles['simple-header__item'])}
              >
                {/* eslint-disable-next-line react/no-unescaped-entities */}
                What's Devesaurus?
              </MenuItem>

              <MenuItem
                item={{
                  url: '/contact',
                  className: styles['simple-header__button'],
                }}
                className={classNames(styles['simple-header__item'])}
              >
                Help Contribute
                <FontAwesomeIcon
                  className={classNames(styles['simple-header__icon'])}
                  icon={['fas', 'heart']}
                />
              </MenuItem>
            </ul>
          </nav>
        </div>
      </header>
    );
  }
}

/* istanbul ignore next */
const mapStateToProps = ({ core: { isMenuOpen } }) => ({
  isMenuOpen,
});

/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  setMenuState: data => dispatch(coreActions.setMenuState(data)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SimpleHeader);
