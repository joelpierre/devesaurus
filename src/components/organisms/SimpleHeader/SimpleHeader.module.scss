@import "~sass/core";

.simple-header {
  $root: &;

  position: relative;
  width: 100%;
  height: auto;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  top: 0;
  left: 0;
  padding: .8rem;
  background-color: $color-tint-alpha;
  transition: $transition-all-base;

  &__wrapper {
    width: 100%;
    display: flex;
  }

  &__hamburger {
    margin-right: auto;
  }

  &__menu {
    margin-left: auto;
    display: inline-flex;
    width: auto;
  }

  &__list {
    width: auto;
    list-style-type: none;
  }

  &__item {
    display: inline-block;
    padding: 0 1rem;
  }

  &__link {
    color: $color-tint-omega;
    text-decoration: none;

    @include hover-focus {
      color: $color-tint-omega;
    }
  }

  &__button {
    border: .2rem solid $color-alpha;
    background-color: $color-alpha;
    border-radius: $radius-rounded;
    color: $color-tint-alpha;
    letter-spacing: .05rem;
    padding: .7rem 1.5rem;
    text-decoration: none;
    font-weight: 600;
    transition: color $transition-base, border-color $transition-base, background-color $transition-base;

    @include hover-focus {
      border-color: $color-tint-omega;
      color: $color-tint-omega;
      text-decoration: none;
      background-color: transparent;

      svg {
        color: red;
      }
    }
  }

  &__icon {
    margin-left: 1rem;
  }
}
