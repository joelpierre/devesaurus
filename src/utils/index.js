export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
