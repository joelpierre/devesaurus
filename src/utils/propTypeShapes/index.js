import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  word: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  syllables: PropTypes.shape({
    count: PropTypes.number.isRequired,
    list: PropTypes.instanceOf(Array).isRequired,
  }).isRequired,
  pronunciation: PropTypes.string,
  results: PropTypes.arrayOf(PropTypes.shape({
    definition: PropTypes.string.isRequired,
    partOfSpeech: PropTypes.oneOf(['noun', 'verb', 'adjective', 'pronoun', 'adverb', 'preposition', 'conjunction', 'interjection']),
  })).isRequired,
  category: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  tags: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  })),
};
