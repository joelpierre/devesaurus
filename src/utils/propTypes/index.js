import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
