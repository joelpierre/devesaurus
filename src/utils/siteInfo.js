import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
