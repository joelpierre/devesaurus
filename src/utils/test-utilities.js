import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';

import rootReducer from '../store/reducers';
// import { sagaMiddleware } from '../store/configureStore';

// /**
//  * Store Factory
//  * @param initialState
//  * @returns {Store<any, Action> & *}
//  */
// export const storeFactory = (initialState) => {
//   const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
//   return createStoreWithMiddleware(rootReducer, initialState);
// };


/**
 * Return node(s) with the given data-test attribute.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper.
 * @param {string} val - Value of the data-test attribute.
 * @returns {ShallowWrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


/**
 * Check the props for a component
 * @param component
 * @param conformingProps
 */
export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name,
  );
  expect(propError)
    .toBeUndefined();
};
