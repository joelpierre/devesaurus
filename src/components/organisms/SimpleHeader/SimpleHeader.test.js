import React from 'react';
import { shallow } from 'enzyme';

import { checkProps, findByTestAttr, matchSnapshot } from '../../../utils/test';

import SimpleHeader from './SimpleHeader';

const defaultProps = {};

/**
 * Factory function to create a ShallowWrapper for the SimpleHeader component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<SimpleHeader {...setupProps} />);
};

describe('<SimpleHeader/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the SimpleHeader Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-simple-header');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(SimpleHeader, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
