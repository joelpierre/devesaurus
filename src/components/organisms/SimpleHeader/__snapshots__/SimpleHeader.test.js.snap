// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleHeader/> should match snapshot 1`] = `
<header
  className="simple-header"
  data-test="component-simple-header"
>
  <div
    className="simple-header__wrapper"
  >
    <Component
      className="simple-header__hamburger"
      descriptor="Menu"
      onClick={[Function]}
    />
    <nav
      className="simple-header__menu"
    >
      <ul
        className="simple-header__list"
      >
        <MenuItem
          className="simple-header__item"
          item={
            Object {
              "className": "simple-header__link",
              "url": "/about",
            }
          }
        >
          What's Devesaurus?
        </MenuItem>
        <MenuItem
          className="simple-header__item"
          item={
            Object {
              "className": "simple-header__button",
              "url": "/contact",
            }
          }
        >
          Help Contribute
          <FontAwesomeIcon
            border={false}
            className="simple-header__icon"
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fas",
                "heart",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </MenuItem>
      </ul>
    </nav>
  </div>
</header>
`;
